### PR TITLE
Add properties to on_deliver hook, so that it can be used also for v3 clients

### PR DESCRIPTION
**Pull request by mths1**
Add properties to on_deliver hook, so that it can be used also for v3 clients

## Proposed Changes

Please describe the big picture of your changes here to communicate to the
VerneMQ team why we should accept this pull request. If it fixes a bug or
resolves a feature request, be sure to link to that issue.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes issue #XXXX)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, styles...)
- [ ] DevOps (Build scripts, pipelines...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [ ] I have read the [CODE_OF_CONDUCT.md](https://github.com/vernemq/vernemq/blob/main/CODE_OF_CONDUCT.md) document
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if needed)
- [ ] Any dependent changes have been merged and published in related repositories
- [ ] I have updated changelog (At the bottom of the release version)
- [ ] I have squashed all my commits into one before merging

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

**Comment by ioolkos** [comment]
@mths1 interesting... what use case do you have in mind?

**Comment by mths1** [comment]
As discussed it is more or less the same use-case as described here already: https://github.com/vernemq/vernemq/issues/1098

**Comment by mths1** [comment]
Seems to be good now. Needs a little change to vernemq_dev before it is really finished.

File: apps/vmq_server/test/vmq_hook_rewrite_SUITE.erl [hunk 1]
```diff
@@ -217,14 +217,14 @@ hook_on_publish_modified_payload(_UserName, _SubscriberId, _QoS, _Topic, <<"rewr
 hook_on_publish_modified_payload(_UserName, _SubscriberId, _QoS, _Topic, Payload, _IsRetain) ->
     throw({expected_payload, <<"rewritten">>, got, Payload}).
 
-hook_on_deliver(_User, {"", <<"dlvr-rewrite-test">>}, [<<"dlvr">>, <<"rewrite">>, <<"payload">>],
-                <<"message">>) ->
+hook_on_deliver(_User, {"", <<"dlvr-rewrite-test">>}, _, [<<"dlvr">>, <<"rewrite">>, <<"payload">>],
```
Comment (ioolkos) [inline apps/vmq_server/test/vmq_hook_rewrite_SUITE.erl:220]: @mths1 We don't necessarily have to change the function arity here. (it's a module hook). Did you see errors from this?

Comment (mths1) [reply apps/vmq_server/test/vmq_hook_rewrite_SUITE.erl:220]: Yep, either vmq_plugin_mgr or the test case starts complaining...

**Review by ioolkos — APPROVED** [review]

**Comment by ioolkos** [comment]
@mths1 do you have a PR for `vernemq_dev`, or should I do it?

**Pull request merged**